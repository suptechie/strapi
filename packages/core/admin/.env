STRAPI_ADMIN_SHOW_TUTORIALS=false
STRAPI_ADMIN_UPDATE_NOTIFICATION=false