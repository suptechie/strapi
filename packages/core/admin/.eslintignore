node_modules/
.eslintrc.js
dist
admin/src/plugins.js