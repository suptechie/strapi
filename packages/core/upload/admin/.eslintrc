{
  "root": true,
  "overrides": [
    {
      "files": ["**/*.js", "**/*.jsx"],
      "extends": ["custom/front"],
      "rules": {
        "import/extensions": "off"
      }
    }
  ]
}