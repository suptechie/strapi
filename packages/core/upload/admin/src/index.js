// NOTE TO PLUGINS DEVELOPERS:
// If you modify this file by adding new options to the plugin entry point
// Here's the file: strapi/docs/3.0.0-beta.x/plugin-development/frontend-field-api.md
// Here's the file: strapi/docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md
// Also the strapi-generate-plugins/files/admin/src/index.js needs to be updated
// IF THE DOC IS NOT UPDATED THE PULL REQUEST WILL NOT BE MERGED
import React from 'react';
import { CheckPagePermissions } from '@strapi/helper-plugin';
import pluginPkg from '../../package.json';
import pluginLogo from './assets/images/logo.svg';
import pluginPermissions from './permissions';
import App from './containers/App';
import Initializer from './containers/Initializer';
import SettingsPage from './containers/SettingsPage';
import InputMedia from './components/InputMedia';
import InputModalStepper from './containers/InputModalStepper';
import reducers from './reducers';
import trads from './translations';
import pluginId from './pluginId';
import { getTrad } from './utils';

const pluginDescription = pluginPkg.strapi.description || pluginPkg.description;
const icon = pluginPkg.strapi.icon;
const name = pluginPkg.strapi.name;

export default {
  register(app) {
    // TODO update doc and guides
    app.addComponents({ name: 'media-library', Component: InputModalStepper });
    // TODO update guide
    app.addFields({ type: 'media', Component: InputMedia });

    app.addReducers(reducers);

    app.registerPlugin({
      description: pluginDescription,
      icon,
      id: pluginId,
      initializer: Initializer,

      isReady: false,
      isRequired: pluginPkg.strapi.required || false,

      mainComponent: App,
      name,
      pluginLogo,
      preventComponentRendering: false,
      settings: {
        global: {
          links: [
            {
              title: {
                id: getTrad('plugin.name'),
                defaultMessage: 'Media Library',
              },
              name: 'media-library',
              to: '/settings/media-library',
              Component: () => (
                <CheckPagePermissions permissions={pluginPermissions.settings}>
                  <SettingsPage />
                </CheckPagePermissions>
              ),
              permissions: pluginPermissions.settings,
            },
          ],
        },
      },
      trads,
      menu: {
        pluginsSectionLinks: [
          {
            destination: `/plugins/${pluginId}`,
            icon,
            label: {
              id: `${pluginId}.plugin.name`,
              defaultMessage: 'Media Library',
            },
            name,
            permissions: pluginPermissions.main,
          },
        ],
      },
    });
  },
  boot() {},
};
