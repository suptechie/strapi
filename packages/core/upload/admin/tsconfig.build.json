{
  "extends": "tsconfig/client.json",
  "compilerOptions": {
    "rootDir": "../",
    "baseUrl": ".",
    "outDir": "./dist"
  },
  "include": ["./src", "./custom.d.ts", "../shared", "../package.json"],
  "exclude": ["**/__mocks__", "./src/**/tests", "**/*.test.*"]
}
