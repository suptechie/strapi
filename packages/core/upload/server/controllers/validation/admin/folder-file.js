'use strict';

const { yup, validateYupSchema } = require('@strapi/utils');

const validateDeleteManyFoldersFilesSchema = yup
  .object()
  .shape({
    fileIds: yup
      .array()
      .min(1)
      .of(yup.strapiID().required()),
    folderIds: yup
      .array()
      .min(1)
      .of(yup.strapiID().required()),
  })
  .noUnknown()
  .required();

module.exports = {
  validateDeleteManyFoldersFiles: validateYupSchema(validateDeleteManyFoldersFilesSchema),
};
