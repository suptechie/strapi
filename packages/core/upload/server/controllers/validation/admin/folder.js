'use strict';

const { yup, validateYupSchema } = require('@strapi/utils');
const { getService } = require('../../../utils');

const NO_SLASH_REGEX = /^[^/]+$/;
const NO_SPACES_AROUND = /^(?! ).+(?<! )$/;

const validateCreateFolderSchema = yup
  .object()
  .shape({
    name: yup
      .string()
      .min(1)
      .matches(NO_SLASH_REGEX, 'name cannot contain slashes')
      .matches(NO_SPACES_AROUND, 'name cannot start or end with a whitespace')
      .required(),
    parent: yup.strapiID().nullable(),
  })
  .noUnknown()
  .required()
  .test('is-folder-unique', 'name already taken', async folder => {
    const { exists } = getService('folder');
    const doesExist = await exists({ parent: folder.parent || null, name: folder.name });
    return !doesExist;
  });

const validateDeleteManyFoldersSchema = yup
  .object()
  .shape({
    ids: yup
      .array()
      .min(1)
      .of(yup.strapiID().required())
      .required(),
  })
  .noUnknown()
  .required();

module.exports = {
  validateCreateFolder: validateYupSchema(validateCreateFolderSchema),
  validateDeleteManyFolders: validateYupSchema(validateDeleteManyFoldersSchema),
};
