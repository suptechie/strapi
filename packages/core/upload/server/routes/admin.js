'use strict';

module.exports = {
  type: 'admin',
  routes: [
    {
      method: 'GET',
      path: '/settings',
      handler: 'upload.getSettings',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'admin::hasPermissions',
            options: {
              actions: ['plugin::upload.settings.read'],
            },
          },
        ],
      },
    },
    {
      method: 'PUT',
      path: '/settings',
      handler: 'upload.updateSettings',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'admin::hasPermissions',
            options: {
              actions: ['plugin::upload.settings.read'],
            },
          },
        ],
      },
    },
    {
      method: 'POST',
      path: '/',
      handler: 'upload.upload',
      config: {
        policies: ['admin::isAuthenticatedAdmin'],
      },
    },
    {
      method: 'GET',
      path: '/files/count',
      handler: 'upload.count',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'admin::hasPermissions',
            options: {
              actions: ['plugin::upload.read'],
            },
          },
        ],
      },
    },
    {
      method: 'GET',
      path: '/files',
      handler: 'upload.find',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'admin::hasPermissions',
            options: {
              actions: ['plugin::upload.read'],
            },
          },
        ],
      },
    },
    {
      method: 'GET',
      path: '/files/:id',
      handler: 'upload.findOne',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'admin::hasPermissions',
            options: {
              actions: ['plugin::upload.read'],
            },
          },
        ],
      },
    },
    {
      method: 'GET',
      path: '/search/:id',
      handler: 'upload.search',
      config: {
        policies: ['admin::isAuthenticatedAdmin'],
      },
    },
    {
      method: 'DELETE',
      path: '/files/:id',
      handler: 'upload.destroy',
      config: {
        policies: [
          'admin::isAuthenticatedAdmin',
          {
            name: 'admin::hasPermissions',
            options: {
              actions: ['plugin::upload.assets.update'],
            },
          },
        ],
      },
    },
  ],
};
