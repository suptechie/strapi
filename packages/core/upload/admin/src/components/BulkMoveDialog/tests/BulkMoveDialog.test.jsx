import React from 'react';

import { NotificationsProvider } from '@strapi/admin/strapi-admin';
import { DesignSystemProvider } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';

import { BulkMoveDialog } from '..';

jest.mock('../../../hooks/useFolderStructure');
jest.mock('../../../hooks/useBulkMove');

const client = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

function ComponentFixture(props) {
  return (
    <QueryClientProvider client={client}>
      <IntlProvider locale="en" messages={{}}>
        <DesignSystemProvider>
          <NotificationsProvider toggleNotification={() => {}}>
            <BulkMoveDialog {...props} />
          </NotificationsProvider>
        </DesignSystemProvider>
      </IntlProvider>
    </QueryClientProvider>
  );
}

function setup(props = { onClose: jest.fn(), selected: [] }) {
  return render(<ComponentFixture {...props} />, { container: document.getElementById('app') });
}

describe('BulkMoveDialog', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders and matches the snapshot', () => {
    setup();
    expect(document.body).toMatchSnapshot();
  });
});
