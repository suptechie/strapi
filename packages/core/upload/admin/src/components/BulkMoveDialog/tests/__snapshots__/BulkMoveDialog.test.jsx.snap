// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkMoveDialog renders and matches the snapshot 1`] = `
.c0 {
  margin-inline-start: -250px;
  position: fixed;
  left: 50%;
  top: 4.6rem;
  z-index: 10;
  width: 50rem;
}

.c3 {
  padding-inline-end: 40px;
  padding-inline-start: 40px;
  position: fixed;
  z-index: 4;
}

.c6 {
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0px 2px 15px rgba(33, 33, 52, 0.1);
  width: 83rem;
}

.c7 {
  background: #f6f6f9;
  padding-block-start: 16px;
  padding-inline-end: 20px;
  padding-block-end: 16px;
  padding-inline-start: 20px;
}

.c11 {
  background: #ffffff;
  padding-block: 8px;
  padding-inline: 8px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
  cursor: pointer;
}

.c14 {
  padding-block: 32px;
  padding-inline: 32px;
}

.c19 {
  display: flex;
}

.c21 {
  padding-inline-end: 12px;
}

.c26 {
  background: #4945ff;
  padding-block: 8px;
  padding-inline: 8px;
  padding-inline-end: 16px;
  padding-inline-start: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c10 {
  font-size: 1.4rem;
  line-height: 1.43;
  font-weight: 600;
  color: #32324d;
}

.c20 {
  font-size: 1.2rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c28 {
  font-size: 1.2rem;
  line-height: 1.33;
  font-weight: 600;
  color: currentcolor;
}

.c1 {
  align-items: stretch;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c4 {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.c9 {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.c18 {
  align-items: stretch;
  display: flex;
  flex-direction: column;
  gap: 4px;
}

.c25 {
  align-items: center;
  display: flex;
  flex-direction: row;
  gap: 8px;
}

.c2 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c12 {
  position: relative;
  outline: none;
}

.c12[aria-disabled='true'] {
  pointer-events: none;
}

.c12:after {
  transition-property: all;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c12:focus-visible {
  outline: none;
}

.c12:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c27 {
  height: 3.2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
  color: #32324d;
}

.c27 svg {
  height: 1.2rem;
}

.c27[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
}

.c27[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
}

.c27:hover {
  background-color: #f6f6f9;
}

.c27:active {
  background-color: #eaeaef;
}

.c29 {
  height: 3.2rem;
  color: #ffffff;
}

.c29 svg {
  height: 1.2rem;
}

.c29[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
}

.c29[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
  color: #666687;
}

.c29:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c29:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c16 {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  gap: 16px;
}

.c17 {
  grid-column: span 12;
  max-width: 100%;
}

.c13 {
  border-color: #dcdce4;
  height: 3.2rem;
  width: 3.2rem;
  color: #8e8ea9;
}

.c13:hover,
.c13:focus {
  color: #666687;
}

.c13[aria-disabled='true'] {
  color: #666687;
}

.c5 {
  background: #32324d1F;
  inset: 0;
}

.c8 {
  border-radius: 4px 4px 0 0;
  border-bottom: 1px solid #eaeaef;
}

.c24 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

.c15 {
  overflow: auto;
  max-height: 60vh;
}

.c22 {
  background: transparent;
  border: none;
  position: relative;
  z-index: 1;
}

.c22 svg {
  height: 1.1rem;
  width: 1.1rem;
}

.c22 svg path {
  fill: #666687;
}

.c23 {
  display: flex;
  background: none;
  border: none;
}

.c23 svg {
  width: 0.9rem;
}

@media (max-width: 1100px) {
  .c17 {
    grid-column: span;
  }
}

@media (max-width: 550px) {
  .c17 {
    grid-column: span 12;
  }
}

<body
  class="lock-body-scroll"
>
  <div>
    <div
      class="c0 c1"
    />
    <span
      class="c2"
    >
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-log"
        role="log"
      />
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-status"
        role="status"
      />
      <p
        aria-live="assertive"
        aria-relevant="all"
        id="live-region-alert"
        role="alert"
      />
    </span>
  </div>
  <div
    class=""
  >
    <div
      class="c3 c4 c5"
    >
      <div>
        <div>
          <div
            aria-labelledby="title"
            aria-modal="true"
            class="c6"
            role="dialog"
          >
            <form
              action="#"
              novalidate=""
            >
              <div
                class="c7 c8"
              >
                <div
                  class="c9"
                >
                  <h2
                    class="c10"
                    id="title"
                  >
                    Move elements to
                  </h2>
                  <button
                    aria-disabled="false"
                    class="c11 c4 c12 c13"
                    type="button"
                  >
                    <span
                      class="c2"
                    >
                      Close the modal
                    </span>
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M26.061 23.939a1.503 1.503 0 0 1-2.125 2.125L16 18.125l-7.939 7.936a1.503 1.503 0 1 1-2.125-2.125L13.875 16 5.939 8.061a1.503 1.503 0 1 1 2.125-2.125L16 13.875l7.939-7.94a1.502 1.502 0 1 1 2.125 2.125L18.125 16z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="c14 c15"
              >
                <div
                  class="c16"
                >
                  <div
                    class="c17"
                  >
                    <div
                      class="c18"
                    >
                      <label
                        alignitems="center"
                        class="c19 c20"
                        for="folder-destination"
                        id="folder-destination-label"
                      >
                        Location
                      </label>
                      <div
                        class=" css-1iyvlxg-container"
                      >
                        <span
                          class="css-1f43avz-a11yText-A11yText"
                          id="react-select-2-live-region"
                        />
                        <span
                          aria-atomic="false"
                          aria-live="polite"
                          aria-relevant="additions text"
                          class="css-1f43avz-a11yText-A11yText"
                          role="log"
                        />
                        <div
                          class=" css-oe0cnm-control"
                        >
                          <div
                            class=" css-onz4xi-ValueContainer"
                          >
                            <div
                              class=" css-1ryi7up-singleValue"
                            >
                              Media Library
                            </div>
                            <input
                              aria-activedescendant=""
                              aria-autocomplete="list"
                              aria-expanded="false"
                              aria-haspopup="true"
                              aria-invalid="false"
                              aria-readonly="true"
                              class="css-mohuvp-dummyInput-DummyInput"
                              id="folder-destination"
                              inputmode="none"
                              role="combobox"
                              tabindex="0"
                              value=""
                            />
                          </div>
                          <div
                            class=" css-1hm8yhz-IndicatorsContainer"
                          >
                            <div
                              aria-hidden="true"
                              class="c21 c22 c23"
                            >
                              <svg
                                fill="currentColor"
                                height="16"
                                viewBox="0 0 32 32"
                                width="16"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                        <input
                          name="destination"
                          type="hidden"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="c7 c24"
              >
                <div
                  class="c9"
                >
                  <div
                    class="c25"
                  >
                    <button
                      aria-disabled="false"
                      class="c26 c25 c12 c27"
                      name="cancel"
                      type="button"
                    >
                      <span
                        class="c28"
                      >
                        Cancel
                      </span>
                    </button>
                  </div>
                  <div
                    class="c25"
                  >
                    <button
                      aria-disabled="false"
                      class="c26 c25 c12 c29"
                      type="submit"
                    >
                      <span
                        class="c28"
                      >
                        Move
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
